spec/helpers/checkout_helper.rb

```diff
@@ -0,0 +1,72 @@
+module CheckoutHelper
+  def add_to_cart(product)
+    visit product
+    click_button 'Add To Cart'
+  end
+
+  def proceed_as_new_user(credentials)
+    fill_inputs(credentials[:email], credentials[:password], credentials[:password])
+
+    click_button 'Create'
+  end
+
+  def proceed_as_guest(credentials)
+    fill_in 'order_email', with: credentials[:email]
+    click_button 'Continue'
+  end
+
+  def fill_in_billing(address)
+    fill_in 'order_bill_address_attributes_firstname', with: address[:first_name]
+    fill_in 'order_bill_address_attributes_lastname', with: address[:last_name]
+    fill_in 'order_bill_address_attributes_address1', with: address[:house_number]
+    fill_in 'order_bill_address_attributes_address2', with: address[:street]
+    fill_in 'order_bill_address_attributes_city', with: address[:city]
+    within('#order_bill_address_attributes_state_id') { select(address[:state]) }
+    fill_in 'order_bill_address_attributes_zipcode', with: address[:zip]
+    fill_in 'order_bill_address_attributes_phone', with: address[:phone]
+  end
+
+  def fill_in_shipping(address)
+    fill_in 'order_ship_address_attributes_firstname', with: address[:first_name]
+    fill_in 'order_ship_address_attributes_lastname', with: address[:last_name]
+    fill_in 'order_ship_address_attributes_address1', with: address[:house_number]
+    fill_in 'order_ship_address_attributes_address2', with: address[:street]
+    fill_in 'order_ship_address_attributes_city', with: address[:city]
+    within('#order_ship_address_attributes_state_id') { select(address[:state]) }
+    fill_in 'order_ship_address_attributes_zipcode', with: address[:zip]
+    fill_in 'order_ship_address_attributes_phone', with: address[:phone]
+  end
+
+  def fill_in_cc(card)
+    fill_in 'card_number', with: card[:number]
+    fill_in 'card_expiry', with: card[:expiry_date]
+    fill_in 'card_code', with: card[:cvv]
+  end
+
+  def save_address(billing_address, shipping_address = nil)
+    fill_in_billing(billing_address)
+
+    if shipping_address
+      uncheck 'order_use_billing'
+      fill_in_shipping(shipping_address)
+    end
+
+    click_button 'Save and Continue'
+  end
+
+  def save_delivery(shipping_method, billing_address = nil, shipping_address = nil)
+    save_address(billing_address, shipping_address) if billing_address
+
+    choose(shipping_method)
+    click_button 'Save and Continue'
+  end
+
+  def save_payment(card, shipping_method = nil, billing_address = nil, shipping_address = nil)
+    save_delivery(shipping_method, billing_address, shipping_address) if shipping_method
```

Comment:
remove this line here, you should use this method before using save_delivery inside the test cases

this one is which we discussed on 2nd floor